src/main/java/com/coveo/pushapiclient/CatalogSource.java

```diff
@@ -0,0 +1,17 @@
+package com.coveo.pushapiclient;
+
+public class CatalogSource {
+
+    public CatalogSource(String apiKey, String organizationId) {
+        throw new java.lang.UnsupportedOperationException("TODO: To implement");
+    }
+
+    public DocumentUpdateService startDocumentUpdate(String string) {
+        throw new java.lang.UnsupportedOperationException("TODO: To implement");
+    }
+
+    public FullCatalogUploadService startfullCatalogUpload(String string) {
```

Comment:
What do you think of naming these methods `update` and `upload/fullUpload`? Alternatively, could be clearer if this was a required constructor argument?

```
new CatalogSource(url, apiKey, mode)
```

Another design could be to have the mode be separate classes that take a `CatalogSource` as a constructor arg, e.g.

```
new UpdateStream(source)

new FullUploadStream(source)
```

Reply:
I like the second approach. It feels weird to tie the mode directly to a source.

Reply:
NVM, the first option really simply things. 👍